src/main/java/net/minestom/arena/game/mob/MobArena.java

```diff
@@ -195,7 +250,24 @@ public void start() {
 
     @Override
     public @NotNull List<Feature> features() {
-        return List.of(Features.combat(), Features.drop());
+        return List.of(Features.combat(false, (attacker, victim) -> {
+            if (attacker instanceof Player player) {
+                final boolean isSword = player.getItemInMainHand()
+                        .material()
+                        .name()
+                        .contains("sword");
```

Comment:
You can use a tag here instead of checking the name. A weapon class tag could be used, which would make it much easier to add new weapons like bows and axes in the future.

Reply:
Changed in the latest commit